### PR TITLE
fix: Remove default implementation of Cache, HttpClient and BeagleLogger 

### LINKED ISSUE #1069: Remove default implementation Cache, HttpClient and BeagleLogger
## Description
Actual these implementations have some problems with security.

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

Closes #1069

### Description and Example

Remove some implementation of core beagle to passing in security test.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
warning_should_call_Log_w_if_is_enable | br.com.zup.beagle.android.logger.BeagleLoggerProxyTest | 0.148|
info_should_call_Log_w_if_is_enable | br.com.zup.beagle.android.logger.BeagleLoggerProxyTest | 0.142|
error_should_call_Log_w_if_is_enable | br.com.zup.beagle.android.logger.BeagleLoggerProxyTest | 0.129|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt [hunk 1]
```diff
@@ -68,21 +68,25 @@ internal class CacheManager(
...
+            }
         }
     }
 
     private fun getBeagleCacheFromDisk(url: String, beagleHashKey: String): BeagleCache? {
         val beagleJsonKey = url.toBeagleJsonKey()
         val beagleTimeKey = url.toBeagleTimeKey()
 
-        val cachedData = storeHandler.restore(StoreType.DATABASE, beagleHashKey, beagleJsonKey, beagleTimeKey)
+        val cachedData = storeHandler!!.restore(StoreType.DATABASE, beagleHashKey, beagleJsonKey, beagleTimeKey)
```
Comment (hernandazevedozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:89]: Is it possible to make this implementation without the `!!` ??

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt [hunk 2]
```diff
@@ -147,7 +151,7 @@ internal class CacheManager(
             jsonKey to responseBody,
             timeKey to nanoTimeInSeconds().toString()
         )
-        storeHandler.save(StoreType.DATABASE, data)
+        storeHandler!!.save(StoreType.DATABASE, data)
```
Comment (hernandazevedozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:154]: Is it possible to make this implementation without the !! ??

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt [hunk 3]
```diff
@@ -156,7 +160,7 @@ internal class CacheManager(
         val cacheMaxRegisters = beagleEnvironment.beagleSdk.config.cache.memoryMaximumCapacity.takeIf { it > 0 }
             ?: beagleEnvironment.beagleSdk.config.cache.size
 
-        val cachedKeys = storeHandler.getAll(StoreType.DATABASE)
+        val cachedKeys = storeHandler!!.getAll(StoreType.DATABASE)
```
Comment (hernandazevedozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:163]: Is it possible to make this implementation without the !! ??

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt [hunk 4]
```diff
@@ -174,7 +178,7 @@ internal class CacheManager(
         val cacheKey = url.toBeagleHashKey()
         val jsonKey = url.toBeagleJsonKey()
         val timeKey = url.toBeagleTimeKey()
-        storeHandler.delete(StoreType.DATABASE, cacheKey)
+        storeHandler!!.delete(StoreType.DATABASE, cacheKey)
```
Comment (hernandazevedozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:181]: Is it possible to make this implementation without the !! ??

**Review by hernandazevedozup — CHANGES_REQUESTED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt (hunk 1 above)
Comment (uziasferreirazup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:89]: No.

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt (hunk 2 above)
Comment (uziasferreirazup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:154]: no.

**Review by hernandazevedozup — APPROVED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt (hunk 1 above)
Comment (matheusribeirozup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:89]: There is always a possibility to replace !!. In this case, I suggest adding any exception other than `KotlinNullPointerException` with a message of what the user should do.
```suggestion
        val cachedData = storeHandler?.restore(StoreType.DATABASE, beagleHashKey, beagleJsonKey, beagleTimeKey) ?: throw NoClassDefFound("Please, provide a StoreHandler implementation")
```

**Review by viniciusguardieirozup — APPROVED** [review]

**Pull request merged**